agent/acs/handler/attach_eni_handler.go

```diff
@@ -0,0 +1,169 @@
+func newAttachENIHandler(ctx context.Context, cluster string, containerInstanceArn string, acsClient wsclient.ClientServer, taskEngine engine.TaskEngine, saver statemanager.Saver) attachENIHandler {
+	// Create a cancelable context from the parent context
+	derivedContext, cancel := context.WithCancel(ctx)
+	return attachENIHandler{
+		messageBuffer:     make(chan *ecsacs.AttachTaskNetworkInterfacesMessage),
+		ctx:               derivedContext,
+		cancel:            cancel,
+		cluster:           aws.String(cluster),
+		containerInstance: aws.String(containerInstanceArn),
+		acsClient:         acsClient,
+		taskEngine:        taskEngine,
+		saver:             saver,
+	}
+}
+
+// handlerFunc returns a function to enqueue requests onto attachENIHandler buffer
+func (attachENIHandler *attachENIHandler) handlerFunc() func(message *ecsacs.AttachTaskNetworkInterfacesMessage) {
+	return func(message *ecsacs.AttachTaskNetworkInterfacesMessage) {
+		attachENIHandler.messageBuffer <- message
+	}
+}
+
+// start invokes handleMessages to ack each enqueued request
+func (attachENIHandler *attachENIHandler) start() {
+	go attachENIHandler.handleMessages()
+}
+
+// stop is used to invoke a cancellation function
+func (attachENIHandler *attachENIHandler) stop() {
+	attachENIHandler.cancel()
+}
+
+// handleMessages handles each message one at a time
+func (attachENIHandler *attachENIHandler) handleMessages() {
+	for {
+		select {
+		case message := <-attachENIHandler.messageBuffer:
+			attachENIHandler.handleSingleMessage(message)
+		case <-attachENIHandler.ctx.Done():
+			return
+		}
+	}
+}
+
+// handleSingleMessage acks the message received
+// TODO: Send response to validate ENI attachment
+func (attachENIHandler *attachENIHandler) handleSingleMessage(message *ecsacs.AttachTaskNetworkInterfacesMessage) error {
+	// Validate fields in the message
+	err := validateAttachTaskNetworkInterfacesMessage(message)
+	if err != nil {
+		return errors.Wrapf(err, "attach eni message handler: error validating AttachTaskNetworkInterfac message received from ECS")
+	}
+
+	err = attachENIHandler.acsClient.MakeRequest(&ecsacs.AckRequest{
+		Cluster:           message.ClusterArn,
+		ContainerInstance: message.ContainerInstanceArn,
+		MessageId:         message.MessageId,
+	})
+	if err != nil {
+		seelog.Warnf("Failed to ack request with messageId: %s, error: %v", aws.StringValue(message.MessageId), err)
+	}
+	attachENIHandler.addENIAttachmentToState(message)
+	err = attachENIHandler.saver.Save()
+	if err != nil {
+		return errors.Wrapf(err, "attach eni message handler: error save agent state")
+	}
+	return nil
+}
+
+// addENIAttachmentToState adds the eni info to the state
+func (handler *attachENIHandler) addENIAttachmentToState(message *ecsacs.AttachTaskNetworkInterfacesMessage) {
+	attachmentArn := aws.StringValue(message.ElasticNetworkInterfaces[0].AttachmentArn)
+	seelog.Info("Adding eni info to state, eni: %s", attachmentArn)
+	handler.taskEngine.AddENIAttachment(
```

Comment:
I don't think that we should use the task engine interface for this. Please use the docker task state interface for this. It's fine if you want to do that in a follow up PR.